Rubberduck.CodeAnalysis/Inspections/Concrete/AssignmentNotUsedInspection.cs

```diff
@@ -108,6 +111,140 @@ private static bool IsAssignmentOfNothing(IdentifierReference reference)
+        /// </remarks>
+        /// <param name="resultCandidate"></param>
+        /// <param name="finder"></param>
+        /// <returns></returns>
+        private static bool IsPotentiallyUsedViaResumeOrGoToExecutionBranch(IdentifierReference resultCandidate, DeclarationFinder finder)
+        {
+            if (!resultCandidate.Declaration.References.Any(rf => !rf.IsAssignment)) { return false; }
+
+            var labelIdLineNumberPairs = finder.DeclarationsWithType(DeclarationType.LineLabel)
+                                                .Where(label => resultCandidate.ParentScoping.Equals(label.ParentDeclaration))
+                                                .Select(lbl => (lbl.IdentifierName, lbl.Context.Start.Line));
+
+            return GotoExecutionBranchPotentiallyUsesVariable(resultCandidate, labelIdLineNumberPairs) 
+                || ResumeExecutionBranchPotentiallyUsesVariable(resultCandidate, labelIdLineNumberPairs);
+        }
+
+        private static bool GotoExecutionBranchPotentiallyUsesVariable(IdentifierReference resultCandidate, IEnumerable<(string, int)> labelIdLineNumberPairs)
+        {
+            var gotoCtxts = resultCandidate.ParentScoping.Context.GetDescendents<VBAParser.GoToStmtContext>()
+                .Where(gotoCtxt => gotoCtxt.Start.Line > resultCandidate.Context.Start.Line);
+
+            if (!gotoCtxts.Any()) { return false; }
+
+            var gotoStmt = GetFirstContextAfterLine(gotoCtxts, resultCandidate.Context.Start.Line);
+
+            if (gotoStmt == null) { return false; }
+
+            var executionBranchLine = DetermineExecutionBranchLine(gotoStmt.expression().GetText(), labelIdLineNumberPairs);
+
+            return IsPotentiallyUsedAssignment(resultCandidate, executionBranchLine);
+        }
+
+        private static bool ResumeExecutionBranchPotentiallyUsesVariable(IdentifierReference resultCandidate, IEnumerable<(string IdentifierName, int Line)> labelIdLineNumberPairs)
+        {
+            var resumeStmtCtxts = resultCandidate.ParentScoping.Context.GetDescendents<VBAParser.ResumeStmtContext>()
+                .Where(jumpCtxt => jumpCtxt.Start.Line > resultCandidate.Context.Start.Line);
+
+            if (!resumeStmtCtxts.Any()) { return false; }
+
+            var onErrorGotoStatements = resultCandidate.ParentScoping.Context.GetDescendents<VBAParser.OnErrorStmtContext>()
+                                .Where(errorStmtCtxt => !errorStmtCtxt.expression().GetText().Equals("0"))
+                                .ToDictionary(k => k.expression()?.GetText() ?? "0", v => v.Start.Line);
+
+            var errorHandlerLabelsAndLines = labelIdLineNumberPairs.Where(pair => onErrorGotoStatements.ContainsKey(pair.IdentifierName));
+            
+            //If the resultCandidate line preceeds all ErrorHandlers/Resume statements - it is not evaluated
+            if (errorHandlerLabelsAndLines.All(s => s.Line > resultCandidate.Context.Start.Line))
```

Comment:
IIUC this is correct, because a line-label needs to be at the start of a line, can you confirm that?

Reply:
From MS-VBAL

> 5.4.1.1 Statement Labels 
>  
> statement-label-definition = LINE-START ((identifier-statement-label ":") / (line-numberlabel [":"] )) 

Admittedly, I often mis-interpret MS-VBAL, but I believe the use of the `LINE-START` terminal symbol establishes that the statement-label-definition begins at the start of a line.  And...I couldn't place them anywhere but at the start of a line in my Excel VBE.